src/docfx/build/toc/TableOfContentsMap.cs

```diff
@@ -164,25 +168,25 @@ public TocCandidate(int subDirectoryCount, int parentDirectoryCount, Document to
         /// <summary>
         /// Compare two toc candidate relative to target file.
         /// Return negative if x is closer than y, possitive if x is farer than y, 0 if x equals y.
-        /// 1. sub nearest
-        /// 2. parent nearest
-        /// 4. sub-name lexicographical nearest
+        /// 1. sub nearest(based on site path)
+        /// 2. parent nearest(based on site path)
+        /// 3. moniker intersection
+        /// 4. file path
         /// </summary>
-        private static int CompareTocCandidate(TocCandidate candidateX, TocCandidate candidateY)
+        private int CompareTocCandidate(TocCandidate candidateX, TocCandidate candidateY, Document file)
         {
-            var subDirCompareResult = candidateX.SubDirectoryCount - candidateY.SubDirectoryCount;
-            if (subDirCompareResult != 0)
+            var result = candidateX.SubDirectoryCount - candidateY.SubDirectoryCount;
+            if (result == 0)
+                result = candidateX.ParentDirectoryCount - candidateY.ParentDirectoryCount;
+            if (result == 0 && _monikerProvider != null)
             {
-                return subDirCompareResult;
+                var fileMonikers = _monikerProvider.GetFileLevelMonikers(file).monikers;
+                result = _monikerProvider.GetFileLevelMonikers(candidateY.Toc).monikers.Intersect(fileMonikers, StringComparer.OrdinalIgnoreCase).Count() -
+                         _monikerProvider.GetFileLevelMonikers(candidateX.Toc).monikers.Intersect(fileMonikers, StringComparer.OrdinalIgnoreCase).Count();
             }
-
-            var parentDirCompareResult = candidateX.ParentDirectoryCount - candidateY.ParentDirectoryCount;
-            if (parentDirCompareResult != 0)
-            {
-                return parentDirCompareResult;
-            }
-
-            return candidateX.Toc.CompareTo(candidateY.Toc);
+            if (result == 0)
```

Comment:
first compare sitePath, then compare monikers?

Reply:
the relative path calculation happens here: https://github.com/dotnet/docfx/pull/4969/files#diff-b15f33cf52e14d45fdb52ca50e899984R127